src/command/render/latexmk/texlive.ts

```diff
@@ -285,14 +286,33 @@ function tlmgrCommand(
   args: string[],
   _quiet?: boolean,
 ) {
+  let cmdExec: string[];
+  let tempFile: string | undefined;
+  if (Deno.build.os === "windows") {
+    // writing to a file to exectute to avoid quoting issue with Deno
+    // https://github.com/quarto-dev/quarto-cli/issues/336
+    const lines = ["tlmgr", cmd, ...args];
+    tempFile = Deno.makeTempFileSync(
+      { prefix: "tlmgr-cmd", suffix: ".bat" },
+    );
+    Deno.writeTextFileSync(tempFile, lines.join(" ") + "\n");
+    cmdExec = ["cmd", "/c", tempFile];
+  } else {
+    cmdExec = ["tlmgr", cmd, ...args];
+  }
   try {
     const result = execProcess(
       {
-        cmd: [...tlmgr, cmd, ...args],
+        cmd: cmdExec,
         stdout: "piped",
         stderr: "piped",
       },
     );
+    result.finally(() => {
```

Comment:
Our typical idiom if we are going to wait on the promise would be to use `await` (and then you can make the finally a normal try/finally statement)

Reply:
Oh I was commenting that below. You were quick on the review.

I tried something like that considering the current try / catch there, but the await is outside as the this function is async for now.  If I add an await it won't be anymore right ? 

This is only to remove the tempfile we are creating, maybe it should be an argument, or maybe it should be written in a place that Quarto is already cleaning up. What the usual workflow with tempfile ? I have looked but not easy to see something useful for this small task.